packages/hierarchies-react/src/presentation-hierarchies-react/LocalizationContext.tsx

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from "react";
+
+/** @internal */
+export interface Localization {
+  loading: string;
+  filterHierarchyLevel: string;
+  clearHierarchyLevelFilter: string;
+  noFilteredChildren: string;
+  resultLimitExceeded: string;
+  resultLimitExceededWithFiltering: string;
+  increaseHierarchyLimit: string;
+  increaseHierarchyLimitWithFiltering: string;
+}
+
+/** @internal */
+export interface LocalizationContext {
+  localization: Localization;
+}
+
+const defaultLocalization: Localization = {
+  loading: "Loading...",
+  filterHierarchyLevel: "Apply filter",
+  clearHierarchyLevelFilter: "Clear active filter",
+  noFilteredChildren: "No child nodes match current filter",
+  resultLimitExceeded: "There are more items than allowed limit of {{limit}}.",
+  resultLimitExceededWithFiltering: "Please provide <link>additional filtering</link> - there are more items than allowed limit of {{limit}}.",
+  increaseHierarchyLimit: "<link>Increase the hierarchy level size limit to {{limit}}.</link>",
+  increaseHierarchyLimitWithFiltering: "Or, <link>increase the hierarchy level size limit to {{limit}}.</link>",
+};
+
+const localizationContext = createContext<LocalizationContext>({ localization: defaultLocalization });
+
+interface LocalizationContextProviderProps {
+  localization?: Localization;
+}
+
+/**
+ * Provides localized strings.
+ * @beta
+ */
+export function LocalizationContextProvider({ localization, children }: PropsWithChildren<LocalizationContextProviderProps>) {
```

Comment:
After some discussions with @saskliutas, we came up with the following plan:

1. This file contains localized strings that are only relevant to iTwinUI-based renderers, so it should be moved into the `itwinui` directory.
2. We believe it's going to be more convenient for consumers to pass localized strings as a prop to the `TreeRenderer` (which is already true), so all this context-related stuff should be `@internal`.
3. The `useTree` and `useUnifiedSelectionTree` hooks are responsible for created a hierarchy provider, which also takes localized strings - we need to ensure consumers can pass them through hook props to the provider.

Reply:
1. Moved the context into `itwinui` directory.
2. Left the context as `@beta` as it should be used to provide localization when `TreeNodeRenderer` is used individually. Added `README` section for it.
3. Added the ability to pass localized string to hierarchy provider.